.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,62 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '34 2 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: [ self-hosted, checkout-default-runners-github-runners ]
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript', 'ruby' ]
```

Comment:
This repo is for iOS which uses Swift as a programming language. Also, we aim to integrate SonarQube on Frames soon. Is having CodeQL mandatory? CC @andra-lezza-cko

Reply:
yes, swift is supported (swift 5.8.1 and above).